support-frontend/assets/helpers/redux/checkout/product/selectors/productPrice.ts

```diff
@@ -0,0 +1,98 @@
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { correspondingPaperProducts } from 'helpers/productPrice/productOptions';
+import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import {
+	finalPrice,
+	getProductPrice,
+} from 'helpers/productPrice/productPrices';
+import { paperProductTypes } from 'helpers/productPrice/subscriptions';
+import type { SubscriptionsState } from 'helpers/redux/subscriptionsStore';
+import type { GuardianProduct, ProductState } from '../state';
+
+function canDeterminePaperPrice(productState: ProductState): boolean {
+	const hasFulfilment = productState.fulfilmentOption !== 'NoFulfilmentOptions';
+	const hasProductOption = productState.productOption !== 'NoProductOptions';
+	return hasFulfilment && hasProductOption;
+}
+
+const requiredFieldsForProduct: Record<
+	GuardianProduct,
+	(productState: ProductState) => boolean
+> = {
+	DigitalPack: () => true,
+	GuardianWeekly: (productState: ProductState) =>
+		productState.fulfilmentOption !== 'NoFulfilmentOptions',
+	Paper: canDeterminePaperPrice,
+	PaperAndDigital: canDeterminePaperPrice,
+	NoProduct: () => false,
+	DailyEdition: () => false,
+	PremiumTier: () => false,
+	ONE_OFF: () => false,
+	MONTHLY: () => false,
+	ANNUAL: () => false,
+};
+
+const getDefaultPriceObject = (currency: IsoCurrency): ProductPrice => ({
+	price: 0,
+	currency,
+	fixedTerm: false,
+});
+
+export function selectPriceForProduct(state: SubscriptionsState): ProductPrice {
```

Comment:
These are the new product price selectors, for selecting the base product price, price with discounts, and the corresponding price with/without digital for the paper checkout.